### DISCUSSION TITLE
Beta discussion

### DISCUSSION THREAD

**Discussion by iboB** [alpaca-core/alpaca-core-sdk#169] [Dev]
Items for beta. They are to be put into the roadmap and made into issues (if not already existing) and beta projects.

* Inference:
	* First and foremost: resolve #5 #151 #166 
	* Schema generation: #149 
	* API features: #47 
	* Llama:
		* [leftover issues](https://github.com/alpaca-core/ac-local/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+label%3Ai%3Allama)
		* Extend API #50
		* Streaming
		* Improve chat
		* RAG
	* Whisper:
		* Improve API, #58 
		* Streaming
	* New inference engines:
		* Some TTS
		* Stable Diffusion
* Tools and demos
	* Acord:
		* WebSocket API
		* Assets and asset coherency (it is a daemon so #96 is not relevant)
		* Some app resembling copilot cli
		* Chat app
	* GUI API browser (ImGui)
	* Performance benchmarks
* Wrappers:
	* Swift: #153 #154
	* Java: some support for non-android gradle builds would be nice
	* NEW: Python
	* NEW: JS/TS (in order of priority)
		* React Native
		* Node modules
		* Browser
		* Low prio and may be abandoned: naked v8, JavaScriptCore... maybe only if using [jsbind](https://github.com/Chobolabs/jsbind)
* Other:
	* CUDA TEE integration
	* Mobile performance stats
	* Improve Android perf (Vulkan experiments)
	
### For other projects

server: server API, web app, #75
	
### Not for Beta. 

Will be discussed post beta:

* Deeper mobile support: #96 #139 
* Training utilities:
	* Federated Learning
	* Reinforcement Learning from Human Feedback
	* Direct Preference Optimization

**Comment by pminev** [comment]
For Stable diffusion we might use [this one](https://github.com/leejet/stable-diffusion.cpp).

**Reply by iboB** [reply]
Yep. This was my intention

**Comment by iboB** [comment]
For TTS, I'm eyeing [tortoise.cpp](https://github.com/balisujohn/tortoise.cpp) but I haven't done thorough research for something better

**Reply by pminev** [reply]
I just found out that bark has [.cpp version](https://github.com/PABannier/bark.cpp). We might add it in future.

**Comment by pminev** [comment]
I've found a model for 3D open-source model called [TripoSR](https://github.com/VAST-AI-Research/TripoSR) which seems interesting.
